app/controllers/peer_reviews_controller.rb

```diff
@@ -157,8 +157,23 @@ def assign_groups
         return
       end
     when 'unassign'
-      deleted_count, undeleted_reviews = PeerReview.unassign(selected_reviewee_group_ids,
-                                                             reviewers_to_remove_from_reviewees_map)
+      unless selected_reviewee_group_ids.empty? # if a row was selected
+        selected_reviewee_group_ids.each do |reviewee_id| # row selected reviewee
```

Comment:
This algorithm is correct but pretty inefficient, as it involves a triply-nested loop.

You can improve this in a few ways:

1. First, by joining on relevant `PeerReview` associations, which in this case are `:reviewer` and `reviewee` (both of these refer to `Grouping`s, but Rails will handle the SQL generation to avoid name conflicts---inspect the logged SQL query for more information)
2. You can use `where` on those associations, e.g. `.where(reviewer: { id: ... })` and again Rails will produce the correct query
3. You can pass in an array of ids to a `where` condition, and the resulting SQL query become an `IN` operation

As usual for complex queries, I recommend experimenting in the Rails console first before writing code to modify this method.